fem/pgridfunc.cpp

```diff
@@ -469,17 +469,17 @@ void ParGridFunction::GetVectorValue(ElementTransformation &T,
    }
 
    Array<int> vdofs;
-   DofTransformation * doftrans = pfes->GetFaceNbrElementVDofs(nbr_el_no,
-                                                               vdofs);
-   const FiniteElement *fe = pfes->GetFaceNbrFE(nbr_el_no);
-
-   int dof = fe->GetDof();
+   DofTransformation * doftrans = pfes->GetFaceNbrElementVDofs(nbr_el_no, vdofs);
    Vector loc_data;
    face_nbr_data.GetSubVector(vdofs, loc_data);
    if (doftrans)
    {
       doftrans->InvTransformPrimal(loc_data);
    }
+
+   const FiniteElement *fe = pfes->GetFaceNbrFE(nbr_el_no);
```

Comment:
Any reason for moving this? I think it makes more sense in the original spot and also is consistent with similar sequences in the rest of this file.

Reply:
When debugging I moved them to the location the variable is first needed whilst looking for segfaults. I generally declare variables as late as possible because a) it's reduces working memory load on the developer whilst improving readability: you don't have to scroll back and forth more to check what the declaration of a variable was, and b) it reduces the number of variables to reason about when bisecting.